### PR TITLE
Make Pointing Device Auto Layer more configurable

### PR DESCRIPTION
**Pull request by drashna**
## Description

As the title says. 

Specifically make the layer timeout and debouncing configurable at runtime.

## Types of Changes

- [x] Core
- [x] Enhancement/optimization
- [x] Documentation

## Issues Fixed or Closed by This PR

* Discord comments in bastardkb #firmware channel

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by Alabastard-64** [comment]
Thanks Drashna I was planning on coming back to this soon to add something
similar but I was also wondering about adding a configurable threshold for
movement as well rather than a simple check for non zero values.

I have had issues with my cursor "vibrating" and activating the automouse
layer if I type a bit too aggressively.

**Comment by drashna** [comment]
I've not experienced that. But might be worth handling that in a generic manner, in the main code as some sort of cursor smoothing option.

File: quantum/pointing_device/pointing_device_auto_mouse.c [hunk 1]
```diff
@@ -20,7 +20,11 @@
 #    include "pointing_device_auto_mouse.h"
 
 /* local data structure for tracking auto mouse */
-static auto_mouse_context_t auto_mouse_context = {.config.layer = (uint8_t)AUTO_MOUSE_DEFAULT_LAYER};
+static auto_mouse_context_t auto_mouse_context = {
+    .config.layer    = (uint8_t)AUTO_MOUSE_DEFAULT_LAYER,
+    .config.timeout  = (uint16_t)AUTO_MOUSE_TIME,
+    .config.debounce = (uint8_t)AUTO_MOUSE_DEBOUNCE,
```
Comment (tzarc) [inline quantum/pointing_device/pointing_device_auto_mouse.c:26]:
```suggestion
    .config.layer    = (uint8_t)(AUTO_MOUSE_DEFAULT_LAYER),
    .config.timeout  = (uint16_t)(AUTO_MOUSE_TIME),
    .config.debounce = (uint8_t)(AUTO_MOUSE_DEBOUNCE),
```

**Review by tzarc — CHANGES_REQUESTED** [review]

File: quantum/pointing_device/pointing_device_auto_mouse.c (hunk 1 above)
Comment (tzarc) [reply quantum/pointing_device/pointing_device_auto_mouse.c:26]: Seemed to revert these in the force-push.

File: docs/feature_pointing_device.md [hunk 2]
```diff
@@ -602,6 +602,10 @@ There are several functions that allow for more advanced interaction with the au
...
 | `auto_mouse_toggle(void)`                                  | Toggle on/off target toggle state (disables layer deactivation when true)            |                           |    `void`(None) |
 | `get_auto_mouse_toggle(void)`                              | Return value of toggling state variable                                              |                           |          `bool` |
+| `set_auto_mouse_timeout(uint16_t timeout)`                 | Change/set the timeout for turing off the layer                                      |                           |    `void`(None) |
+| `get_auto_mouse_timeout(void)`                             | Return the current timeout for turing off the layer                                  |                           |      `uint16_t` |
+| `set_auto_mouse_debounce(uint16_t timeout)`                | Change/set the debounce for preventing layer activation                              |                           |    `void`(None) |
+| `get_auto_mouse_debounce(void)`                            | Return the current debounce for preventing layer activation                          |                           |      `uint8_t` |
```
Comment (elpekenin) [inline docs/feature_pointing_device.md:608]: Mis-aligned
```suggestion
| `get_auto_mouse_debounce(void)`                            | Return the current debounce for preventing layer activation                          |                           |       `uint8_t` |
```

**Review by tzarc — APPROVED** [review]

**Pull request merged**